Lib/test/test_format.py

```diff
@@ -492,6 +492,14 @@ def test_precision_c_limits(self):
         c = complex(f)
         with self.assertRaises(ValueError) as cm:
             format(c, ".%sf" % (INT_MAX + 1))
+            
+    def test_issue_4588_rustpython(self):
+        x = 123456789012345678901234567890
+        for i in range(0, 30):
+            format(x, ',')
+
+            # move to the next integer to test
+            x = x // 10
```

Comment:
we don't add our original tests under `Lib/`. Could you move this test under `extra_tests/snippets/`?